src/bacnet/basic/object/iv.c

```diff
@@ -221,20 +279,52 @@ bool Integer_Value_Present_Value_Set(
 bool Integer_Value_Object_Name(
     uint32_t object_instance, BACNET_CHARACTER_STRING *object_name)
 {
-    char text_string[32] = "";
-    unsigned int index;
     bool status = false;
+    unsigned index = 0;
+
+    if (!object_name) {
+        return false;
+    }
 
     index = Integer_Value_Instance_To_Index(object_instance);
-    if (index < MAX_INTEGER_VALUES) {
-        sprintf(
-            text_string, "INTEGER VALUE %lu", (unsigned long)object_instance);
-        status = characterstring_init_ansi(object_name, text_string);
+    if (index < IV_Max_Index) {
+        *object_name = IV_Descr[index].Name;
+        status = true;
     }
 
     return status;
 }
 
+/**
+ * For a given object instance-number, return the description.
+ *
+ * Note: the object name must be unique within this device.
+ *
+ * @param  object_instance - object-instance number of the object
+ * @param  description - description pointer
+ *
+ * @return  true/false
+ */
+bool Integer_Value_Description(
+    uint32_t object_instance, BACNET_CHARACTER_STRING *description)
+{
+    bool status = false;
+    unsigned index = 0;
+
+    if (!description) {
+        return false;
+    }
+
+    index = Integer_Value_Instance_To_Index(object_instance);
+    if (index < IV_Max_Index) {
+        *description = IV_Descr[index].Description;
+        status = true;
```

Comment:
well status is not used otherwise we could get a rid of it and return true right away